zulipterminal/model.py

```diff
@@ -660,14 +661,49 @@ def get_user_info(self, user_id: int) -> Dict[str, Any]:
             user_info['Role'] = next((r for r, is_role in user_role.items()
                                       if is_role), 'Member')
 
+            # * This logic should match `user_last_seen_time_status` in
+            # * the web app's
+            # * `static/js/buddy_data.js`.
             if self.initial_data['presences'].get(data['email'], None):
-                presence = (self.initial_data['presences'][data['email']]
-                            ['aggregated']['timestamp'])
-                user_info['Last active'] = (datetime.fromtimestamp(presence)
-                                            .strftime("%d-%b-%Y %H:%M:%S"))
+                user_info['Last active'] = self.get_status(data['email'])
+            else:
+                # There are situations where the client has incomplete
+                # presence history on a user. This can happen when users are
+                # deactivated, or when they just haven't been present in a
+                # long time (and we may have queries on presence that go back
+                # only N weeks).
+                # We give the somewhat vague status of "Unknown" for these
+                # users.
+                user_info['Last active'] = 'More than 2 weeks ago'
```

Comment:
This doesn't seem to match; could you explain?

Reply:
Ah, that needs to be amended. Actually, I picked that whole comment (L670-L676) while referring to the web app's implementation :sweat_smile: 

All that means (AFAIK) is that the Webapp declares the status `Active More than two weeks ago` for all those users whose either account is being deactivated, or are not active for a very long time :)

Reply:
Does the webapp implementation use that text, or "Unknown"?

Reply:
I guess they used it just to symbolize that the data of a user inactive for more than 2 weeks ago is undefined.

[This](https://chat.zulip.org/#narrow/stream/9-issues/topic/Last.20Online.3A.202.20Weeks.20Ago/near/727386) message on czo might be useful

Reply:
Going as per the webapp seems good - and referencing that discussion or the issue may be relevant too.